Framework/PythonInterface/plugins/algorithms/SNSPowderReduction.py

```diff
@@ -344,6 +365,10 @@ def PyExec(self):  # noqa
         if self.COMPRESS_TOL_TOF < -0.:
             self.COMPRESS_TOL_TOF = 0.01
 
+        # Clean the cache directory if so requested
+        if self._clean_cache:
+            api.CleanFileCache(CacheDir=self._cache_dir, AgeInDays=0)
```

Comment:
Force remove everything sounds really dangerous here, especially if someone accidentally specify the raw data directory as the CacheDir.
I think we should have some safe guard here to make sure that `CacheDir` does not contain any raw data.
Perhaps discussing this with the CIS can help clarify things a little bit.

Reply:
`CleanFileCache` will remove files matching one of these two patterns:
```python
".*_[0-9a-f]{40}.nxs"
"[0-9a-f]{40}.nxs"
```
It will be highly unlikely that a user has a file containing a single word, 40 characters long, made up only of numbers and lower-case letters. I think we're safe here.